src/masternodeman.cpp

```diff
@@ -136,17 +136,32 @@ void CMasternodeMan::AskForMN(CNode* pnode, const CTxIn &vin)
 {
     if(!pnode) return;
 
-    std::map<COutPoint, int64_t>::iterator it = mWeAskedForMasternodeListEntry.find(vin.prevout);
-    if (it != mWeAskedForMasternodeListEntry.end() && GetTime() < (*it).second) {
-        // we've asked recently, should not repeat too often or we could get banned
-        return;
-    }
+    LOCK(cs);
 
-    // ask for the mnb info once from the node that sent mnp
+    std::map<COutPoint, std::map<CNetAddr, int64_t> >::iterator it1 = mWeAskedForMasternodeListEntry.find(vin.prevout);
```

Comment:
You copy the map<CNetAddr,int64_t> into tmpMap, modify it then copy it back again.  You could just modify it in place.